atom_batch_execution/scripts/plot_graphs

```diff
@@ -0,0 +1,325 @@
+
+        if all(custom_legend):
+            custom_legend = eval(plot_options.get('custom_legend'))
+
+        # Normalizing options to allow them to be defined as lists or individual str/int
+        #Strs
+        if isinstance(colors,str):
+            print(f"Colors {Fore.BLUE}{colors}{Style.RESET_ALL} set as a single {Fore.BLUE}str{Style.RESET_ALL}, expanding to all plots in this figure.")
+            colors = [colors for _ in range(number_of_lines)]
+
+        if isinstance(linestyles,str):
+            print(f"Linestyles {Fore.BLUE}{linestyles}{Style.RESET_ALL} set as a single {Fore.BLUE}str{Style.RESET_ALL}, expanding to all plots in this figure.")
+            linestyles = [linestyles for _ in range(number_of_lines)]
+
+        if isinstance(markers,str):
+            print(f"Markers {Fore.BLUE}{markers}{Style.RESET_ALL} set as a single {Fore.BLUE}str{Style.RESET_ALL}, expanding to all plots in this figure.")
+            markers = [markers for _ in range(number_of_lines)]
+
+        #Ints
+        if isinstance(markersizes,int):
+            print(f"Markersizes {Fore.BLUE}{markersizes}{Style.RESET_ALL} set as a single {Fore.BLUE}int{Style.RESET_ALL}, expanding to all plots in this figure.")
+            markersizes = [markersizes for _ in range(number_of_lines)]
+
+        if isinstance(linewidths,int):
+            print(f"Linewidths {Fore.BLUE}{linewidths}{Style.RESET_ALL} set as a single {Fore.BLUE}int{Style.RESET_ALL}, expanding to all plots in this figure.")
+            linewidths = [linewidths for _ in range(number_of_lines)]
+
+
+        plt.figure()
+        for idx,plot_line in enumerate(plots_configs[plot_figure]['data']):
+
+            x_values = plot_line['x_values']
+            y_values = plot_line['y_values']
+            legend   = plot_line.get('legend')
+
+            # Plotting
+            plt.plot(x_values, y_values, color=colors[idx], linestyle=linestyles[idx], linewidth=linewidths[idx],
+                     markersize = markersizes[idx],marker = markers[idx],label = legend)
+            # Might want to be ommited to be replaced by a caption in papers
+            if not hide_title:
+                plt.title(title)
+
+            plt.xlabel(xlabel)
+            plt.ylabel(ylabel)
+            plt.grid(grid)
+            plt.xscale(xscale)  # Set x-scale
+            plt.yscale(yscale)  # Set y-scale
+
+            # from matplotlib.ticker import MaxNLocator, LogLocator,FormatStrFormatter,ScalarFormatter
+            # import numpy as np
+            # plt.gca().yaxis.set_major_locator(LogLocator(base=10, subs=np.arange(0.1, 1, 0.1), numticks=10))
+            # # plt.gca().yaxis.set_major_formatter(FormatStrFormatter(''))
+            # plt.gca().yaxis.set_major_formatter(ScalarFormatter())
+            # plt.gca().yaxis.set_minor_formatter(ScalarFormatter())
+            # plt.tick_params(axis='y', labelsize=8)  # Adjust the label size here (e.g., 8)
```

Comment:
Please refrain from having big blocks of code commented

Reply:
This code was to have the plot use a custom Log scale. I spent a considerable amount of time getting it to work and its there to eventually be implemented in a clean way. This script is still experimental so I think this comment should stay in the code. I will create a low priority issue and tag it there.